pkg/openstackclient/errors.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2022 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openstackclient
+
+import (
+	"errors"
+	"fmt"
+
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// ForbiddenPodSpecChange - error returned when Pod must be recreate
+type ForbiddenPodSpecChange struct {
+	Field string
+}
+
+func (e *ForbiddenPodSpecChange) Error() string {
+	return fmt.Sprintf("cannot update Pod spec field: %s", e.Field)
+}
+
+// IsForbiddenPodSpecChange -
+func IsForbiddenPodSpecChange(err error) bool {
+	return errors.Is(err, &ForbiddenPodSpecChange{}) || k8s_errors.IsInvalid(err)
+}
```

Comment:
could also add all this to common/pod.go as its generic and could be used in future for other pod handling?

Reply:
Or create common/errors.go?

Reply:
either is fine for me, I just think it might be good in the common space

Reply:
Yes, I was thinking the same but I figured we could come back to it later if/when we add more error type (this is the first one, IIRC).

Reply:
I'll create common/errors.go as this seems to be the convention (e.g https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors)